src/ecalc/libraries/libecalc/common/libecalc/input/mappers/model.py

```diff
@@ -66,12 +66,31 @@ def _get_curve_data_from_resource(resource: Resource, speed: float = 0.0):
 def _single_speed_compressor_chart_mapper(model_config: Dict, resources: Resources) -> dto.SingleSpeedChart:
     units = get_units_from_chart_config(chart_config=model_config)
     curve_config: Optional[Dict] = model_config.get(EcalcYamlKeywords.consumer_chart_curve)
+    name = model_config.get(EcalcYamlKeywords.name)
+    curves_instead_of_curve = (
+        True if isinstance(model_config.get(EcalcYamlKeywords.consumer_chart_curves), dict) else False
+    )
+
+    # Check if user has used CURVES (reserved for variable speed compressors)
+    # instead of CURVE (should be used for single speed compressors),
+    # and give clear error message.
+    if curves_instead_of_curve:
+        raise DataValidationError(
+            data=model_config,
+            message=f"Compressor model {name}:\n"
+            f"The keyword {EcalcYamlKeywords.consumer_chart_curves} should only be used for "
+            f"variable speed compressor models.\n"
+            f"{name} is a single speed compressor model and should use the keyword "
+            f"{EcalcYamlKeywords.consumer_chart_curve}.",
+        )
 
     if not isinstance(curve_config, dict):
         raise DataValidationError(
             data=model_config,
-            message=f"{EcalcYamlKeywords.consumer_chart_curve}"
-            f" should be an object. Type given: {type(curve_config)}.",
+            message=f"The keyword {EcalcYamlKeywords.consumer_chart_curve} is not specified "
```

Comment:
The check `if EcalcYamlKeywords.consumer_chart_curve not in model_config:` would make more sense here, we could move it up to the start of the function, then remove Optional on the curve_config type.

Reply:
If the check `if isinstance(curve_config, dict):` is replaced with `if EcalcYamlKeywords.consumer_chart_curve not in model config:`, mypy fails with the following error:

> Unsupported right operand type for in ("Any | None")

Reply:
Seems like it is an issue with mypy (it doesn´t apply type narrowing within nested functions). Adding `# type: ignore` after the next check (`if EcalcYamlKeywords.file in curve_config:`) solves it...

Reply:
Does it help to add a check for `if not isinstance(model_config, dict)` then raise a relevant error? Not sure why mypy complains when we have specified dict as type for model_config

Reply:
Yes it helps.